tests/test_kaggle_module_resolver.py

```diff
@@ -2,62 +2,100 @@
+    addr = f"http://{hostname}:{port}"
 
-            with open('/input/tests/data/model.tar.gz', 'rb') as model_archive:
-                self.wfile.write(model_archive.read())
+    # Simulates we are inside a Kaggle environment.
+    env = EnvironmentVarGuard()
+    env.set('KAGGLE_KERNEL_RUN_TYPE', 'Interactive')
+    env.set('KAGGLE_USER_SECRETS_TOKEN', 'foo jwt token')
+    env.set('KAGGLE_DATA_PROXY_TOKEN', 'foo proxy token')
+    env.set('KAGGLE_DATA_PROXY_URL', addr)
 
-    def _test_client(self, client_func, handler):
-        with HTTPServer(('localhost', 8080), handler) as test_server:
+    with env:
+        with HTTPServer((hostname, port), handler_class) as test_server:
             threading.Thread(target=test_server.serve_forever).start()
 
             try:
-                client_func()            
+                yield addr
             finally:
                 test_server.shutdown()
 
-    def test_kaggle_resolver_succeeds(self):
-        # Simulates we are inside a Kaggle environment.
-        env = EnvironmentVarGuard()
-        env.set('KAGGLE_CONTAINER_NAME', 'foo')
-        # Attach model to right directory.
-        os.makedirs('/kaggle/input/foomodule/tensorflow2/barvar')
-        os.symlink('/input/tests/data/saved_model/', '/kaggle/input/foomodule/tensorflow2/barvar/2', target_is_directory=True)
+class HubHTTPHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200)
+        self.send_header('Content-Type', 'application/gzip')
+        self.end_headers()
+
+        with open('/input/tests/data/model.tar.gz', 'rb') as model_archive:
+            self.wfile.write(model_archive.read())
 
-        with env:
+class KaggleJwtHandler(BaseHTTPRequestHandler):
+    def do_POST(self):
+        if self.path.endswith("AttachDatasourceUsingJwtRequest"):
+            content_length = int(self.headers["Content-Length"])
+            request = json.loads(self.rfile.read(content_length))
+            model_ref = request["modelRef"]
+
+            self.send_response(200)
+            self.send_header("Content-type", "application/json")
+            self.end_headers()
+
+            if model_ref['ModelSlug'] == 'unknown':
+                self.wfile.write(bytes(json.dumps({
+                    "wasSuccessful": False,
+                }), "utf-8"))
+                return
+
+            # Load the files
+            mount_slug = f"{model_ref['ModelSlug']}/{model_ref['Framework']}/{model_ref['InstanceSlug']}/{model_ref['VersionNumber']}"
+            os.makedirs(os.path.dirname(os.path.join(MOUNT_PATH, mount_slug)))
+            os.symlink('/input/tests/data/saved_model/', os.path.join(MOUNT_PATH, mount_slug), target_is_directory=True)
+
+            # Return the response
+            self.wfile.write(bytes(json.dumps({
+                "wasSuccessful": True,
+                "result": {
+                    "mountSlug": mount_slug,
+                },
+            }), "utf-8"))
+        else:
```

Comment:
nit: start with this case to reduce tabs:
```python
if not self.path.endswith(...):
  self.send_response(404)
  ...
content_length = int(self.headers["Content-Length"])
...
```

Reply:
I prefer to keep it this way because we may add new `elif` cases for other paths and we would want to fail only in the `else` at the bottom.